phoenix-core/src/test/java/org/apache/phoenix/cache/ServerMetadataCacheIT.java

```diff
@@ -891,12 +891,14 @@ public void testSelectQueryAfterAlterIndex() throws Exception {
             stmt.executeQuery("SELECT k FROM " + tableName + " WHERE v1=1");
             Assert.assertEquals("Query on secondary key should have used index.", indexName, stmt.getQueryPlan().getTableRef().getTable().getTableName().toString());
 
-            //verify client-2 cache was updated with the index's base table metadata
```

Comment:
change needed because earlier I was calling MetadataClient.updateCache for the parent table but the correct way is to call it for the index.

Reply:
> change needed because earlier I was calling MetadataClient.updateCache for the parent table but the correct way is to call it for the index.

Where is this change being made?  @palashc

Reply:
PhoenixStatement.java, there are only 3 changes in this PR